test/setup.spec.js

```diff
@@ -466,6 +467,107 @@ describe('setup', () => {
       .then(done)
   })
 
+  it('should unwrap on the template', () => {
+    const vm = new Vue({
+      setup() {
+        const r = ref('r')
+        const nested = {
+          a: ref('a'),
+          aa: {
+            b: ref('aa'),
+            bb: {
+              cc: ref('aa'),
+              c: 'aa',
+            },
+          },
+
+          aaa: reactive({
+            b: ref('aaa'),
+            bb: {
+              c: ref('aaa'),
+              cc: 'aaa',
+            },
+          }),
+
+          aaaa: {
+            b: [1],
+            bb: ref([1]),
+            bbb: reactive({
+              c: [1],
+              cc: ref([1]),
+            }),
+            bbbb: [ref(1)],
+          },
+        }
+
+        const refList = ref([ref('1'), ref('2'), ref('3')])
+        const list = [ref('a'), ref('b')]
+
+        return {
+          r,
+          nested,
+          refList,
+          list,
+        }
+      },
+      template: `<div>
+        <p id="r">{{r}}</p>
+        <p id="nested">{{nested.a}}</p>
+        <p id="list">{{list}}</p>
+        <p id="refList">{{refList}}</p>
+
+        <p id="nested_aa_b">{{ nested.aa.b }}</p>
+        <p id="nested_aa_bb_c">{{ nested.aa.bb.c }}</p>
+        <p id="nested_aa_bb_cc">{{ nested.aa.bb.cc }}</p>
+
+        <p id="nested_aaa_b">{{ nested.aaa.b }}</p>
+        <p id="nested_aaa_bb_c">{{ nested.aaa.bb.c }}</p>
+        <p id="nested_aaa_bb_cc">{{ nested.aaa.bb.cc }}</p>
+
+        <p id="nested_aaaa_b">{{ nested.aaaa.b }}</p>
+        <p id="nested_aaaa_bb_c">{{ nested.aaaa.bb }}</p>
+        <p id="nested_aaaa_bbb_cc">{{ nested.aaaa.bbb.c }}</p>
+        <p id="nested_aaaa_bbb_cc">{{ nested.aaaa.bbb.cc }}</p>
```

Comment:
same id, different ref, and confused with :528 which shouldn't unwrap - the tests for those 3 cases are wrong (overwriting one another)

Reply:
Nice catch. Would you mind open a PR? Thanks.